core/src/main/scala/persistentscheduler/PersistentScheduler.scala

```diff
@@ -78,29 +82,30 @@ class PersistentScheduler(persistence: SchedulerPersistence) extends Actor
     self ! State(subscriptions)
   }
 
-  def schedulePersistenceCheckForEvents(): Unit = {
+  def schedulePersistenceCheckForEvents(): Cancellable = {
     scheduler.schedule(1.minute, 1.minute, self, CheckPersistenceForEvents())
   }
 
-  def scheduleNextEventFromPersistence(): Unit = {
+  def scheduleNextEventFromPersistence(): Option[Cancellable] = {
     nextCancellable.filter(!_.isCancelled).foreach(_.cancel())
 
     nextEvent = persistence.next(1).asScala.headOption
     nextCancellable = nextEvent.map(schedule(_))
+
+    nextCancellable
   }
 
-  def publishEvent(e: TimedEvent): Unit = {
+  def publishEvent(e: TimedEvent): Set[Future[Any]] = {
 
     persistence.delete(e.id)
```

Comment:
shouldn't actually all methods of the `SchedulerPersistence.scala` return Future[<type>] because this is the thing that operates with the DB and blocks?